src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```
@@ -365,8 +346,45 @@ else if (hasFailures && testCase.getFailures().get(0).getType() != null) {
+     * @return a list of helpful messages that can be added to an automatic {@link Feedback}.
+     */
+    private List<String> extractMessagesFromTestCase(final TestCaseDTO testCase) {
+        var hasErrors = !CollectionUtils.isEmpty(testCase.getErrors());
+        var hasFailures = !CollectionUtils.isEmpty(testCase.getFailures());
+        var hasSuccessInfos = !CollectionUtils.isEmpty(testCase.getSuccessInfos());
+        boolean successful = testCase.isSuccessful();
+
+        final List<String> feedbackMessages = new ArrayList<>();
+
+        if (successful && hasSuccessInfos && testCase.getSuccessInfos().get(0).getMostInformativeMessage() != null) {
+            feedbackMessages.add(testCase.getSuccessInfos().get(0).getMostInformativeMessage());
+        }
+        else if (hasErrors && testCase.getErrors().get(0).getMostInformativeMessage() != null) {
+            feedbackMessages.add(testCase.getErrors().get(0).getMostInformativeMessage());
+        }
+        else if (hasFailures && testCase.getFailures().get(0).getMostInformativeMessage() != null) {
+            feedbackMessages.add(testCase.getFailures().get(0).getMostInformativeMessage());
+        }
+        else if (hasErrors && testCase.getErrors().get(0).getType() != null) {
+            feedbackMessages.add(String.format("Unsuccessful due to an error of type: %s", testCase.getErrors().get(0).getType()));
+        }
+        else if (hasFailures && testCase.getFailures().get(0).getType() != null) {
+            feedbackMessages.add(String.format("Unsuccessful due to an error of type: %s", testCase.getFailures().get(0).getType()));
+        }
+        else if (!successful) {
+            // this is an edge case which typically does not happen
+            feedbackMessages.add("Unsuccessful due to an unknown error. Please contact your instructor!");
+        }
+        // default case: successful and no message available => do not generate one
```

default might be misleading here. I guess you mean "default" in the sense of a switch statement, but people could think that this case happens 99% of the time and all other cases are edge cases (which is not true)